vars/runIntegTestScript.groovy

```diff
@@ -12,10 +12,19 @@ void call(Map args = [:]) {
     String jobName = args.jobName ?: 'distribution-build-opensearch'
     def buildManifest = lib.jenkins.BuildManifest.new(readYaml(file: args.buildManifest))
 
-    echo "Start integTest for distribution type: " + buildManifest.getDistribution()
+    String architecture = buildManifest.getArtifactArchitecture()
+    String distribution = buildManifest.getDistribution()
+    String platform = buildManifest.getArtifactPlatform()
+    echo "Start integTest on: " + distribution + " " + architecture + " " + platform
 
-    javaVersion = (jobName.equals('distribution-build-opensearch')) ? detectTestDockerAgent(testManifest: args.testManifest).javaVersion : ''
+    def javaVersion = (jobName.equals('distribution-build-opensearch')) ? detectTestDockerAgent(testManifest: args.testManifest).javaVersion : ''
     String javaHomeCommand = (jobName.equals('distribution-build-opensearch') && ! javaVersion.equals('')) ? "JAVA_HOME=/opt/java/${javaVersion}" : ''
+    if (platform.equals('windows')) { // Windows use scoop to switch the Java Version
+        String javaVersionNumber = javaVersion.replaceAll("[^0-9]", "") // Only get number
+        echo("Switching to Java ${javaVersionNumber} on Windows Docker Container")
+        sh("scoop reset `scoop list jdk | grep ${javaVersionNumber} | head -1 | cut -d ' ' -f1`")
+        javaHomeCommand = ''
```

Comment:
So we are resetting `JAVA_HOME` to empty string in case of windows?

Reply:
Yes, Windows is on a completely different situation, where scoop is handling the setups of jdk and java_home.
We reset javahomecommand so it will not mess with scoop settings.

It is worth an improvement on the entire logic later on.